ironic/drivers/modules/drac/inspect.py

```diff
@@ -27,24 +29,137 @@
+        else:
+            LOG.warning("No port information discovered "
+                        "for node %(node)s", {'node': node.uuid})
+
+        return states.MANAGEABLE
+
+    def _get_pxe_port_macs(self, task):
+        """Get a list of pxe port MAC addresses.
+
+        :param task: a TaskManager instance.
+        :returns: Returns list of pxe port MAC addresses.
+        """
+        system = redfish_utils.get_system(task.node)
+
+        pxe_port_macs = []
+        pxe_params = ["PxeDev1EnDis", "PxeDev2EnDis",
+                      "PxeDev3EnDis", "PxeDev4EnDis"]
+        pxe_nics = ["PxeDev1Interface", "PxeDev2Interface",
+                    "PxeDev3Interface", "PxeDev4Interface"]
+
+        redfish_ip = task.node.driver_info.get('redfish_address')
+        redfish_username = task.node.driver_info.get('redfish_username')
+        redfish_password = task.node.driver_info.get('redfish_password')
+
+        # Get dictionary of ethernet interfaces
+        ethernet_interfaces = system.ethernet_interfaces.get_members()
+        ethernet_interfaces_mac_dict = {interface.identity: interface.mac_address for interface in ethernet_interfaces}
+
+        if system.boot.mode == 'uefi':
+            for param, nic in zip(pxe_params, pxe_nics):
+                if system.bios.attributes[param] == 'Enabled':
+                    nic_id = system.bios.attributes[nic]
+                    # Get MAC address of the given nic_id
+                    mac_address = ethernet_interfaces_mac_dict[nic_id]
+                    pxe_port_macs.append(mac_address)
+        elif system.boot.mode == 'bios':
+            # Get instance of OEM extension
+            authenticator = sushy.auth.BasicAuth(redfish_username, redfish_password)
+            url = '%s%s' % (redfish_ip, _SERVICE_ROOT)
+            conn = sushy.Sushy(url, verify=False, auth=authenticator)
+            manager = conn.get_manager('iDRAC.Embedded.1')
+            oem_manager = manager.get_oem_extension('Dell')
+            job_response = oem_manager.export_system_configuration(target="NIC")
+
+            if job_response.status_code != _JOB_RESPONSE_CODE:
+                error = (_('An error occurred when attempting to export '
+                           'the system configuration. Status code: %(code), '
+                           'Error details: %(err)'),
+                         {'code': job_response.status_code,
+                          'err': job_response.__dict__})
+                LOG.error(error)
+                raise exception.HardwareInspectionFailure(error=error)
+
+            if _SYSTEM_CONFIG_TAG in str(job_response.__dict__):
+                xml_string = job_response.__dict__.get('_content')
+                root = et.fromstring(xml_string)
+                for child in root:
+                    nic_id = [child.attrib.get('FQDD') for ch in child if
+                              ch.text == 'PXE']
```

Comment:
Can we be a little more discriminate when looking for PXE enabled ports above?  We should be looking for an attribute with a Name of LegacyBootProto, and then checking to see if it is set to PXE or not.  And there should be only 1 of these in a given child, so there should be no need to make an array of them.